incubator/metrics-http/src/main/java/io/aklivity/zilla/runtime/metrics/http/internal/HttpActiveRequestsMetricContext.java

```diff
@@ -0,0 +1,125 @@
+ * License at
+ *
+ *   https://www.aklivity.io/aklivity-community-license/
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package io.aklivity.zilla.runtime.metrics.http.internal;
+
+import static io.aklivity.zilla.runtime.engine.metrics.MetricContext.Direction.BOTH;
+import static io.aklivity.zilla.runtime.metrics.http.internal.HttpUtils.commonId;
+import static io.aklivity.zilla.runtime.metrics.http.internal.HttpUtils.streamDirection;
+
+import java.util.function.LongConsumer;
+
+import org.agrona.DirectBuffer;
+import org.agrona.collections.Long2LongHashMap;
+
+import io.aklivity.zilla.runtime.engine.binding.function.MessageConsumer;
+import io.aklivity.zilla.runtime.engine.metrics.Metric;
+import io.aklivity.zilla.runtime.engine.metrics.MetricContext;
+import io.aklivity.zilla.runtime.metrics.http.internal.types.stream.AbortFW;
+import io.aklivity.zilla.runtime.metrics.http.internal.types.stream.BeginFW;
+import io.aklivity.zilla.runtime.metrics.http.internal.types.stream.EndFW;
+import io.aklivity.zilla.runtime.metrics.http.internal.types.stream.FrameFW;
+import io.aklivity.zilla.runtime.metrics.http.internal.types.stream.ResetFW;
+
+public final class HttpActiveRequestsMetricContext implements MetricContext
+{
+    private final String group;
+    private final Metric.Kind kind;
+    private final FrameFW frameRO = new FrameFW();
+
+    public HttpActiveRequestsMetricContext(
+        String group,
+        Metric.Kind kind)
+    {
+        this.group = group;
+        this.kind = kind;
+    }
+
+    @Override
+    public String group()
+    {
+        return group;
+    }
+
+    @Override
+    public Metric.Kind kind()
+    {
+        return kind;
+    }
+
+    @Override
+    public Direction direction()
+    {
+        return BOTH;
+    }
+
+    @Override
+    public MessageConsumer supply(
+        LongConsumer recorder)
+    {
+        return new HttpActiveRequestsMetricHandler(recorder);
+    }
+
+    private final class HttpActiveRequestsMetricHandler implements MessageConsumer
+    {
+        private static final long INITIAL_STATUS = 0b11L;
+
+        private final LongConsumer recorder;
+        private final Long2LongHashMap streams = new Long2LongHashMap(0L);
+
+        private HttpActiveRequestsMetricHandler(
+            LongConsumer recorder)
+        {
+            this.recorder = recorder;
+        }
+
+        @Override
+        public void accept(
+            int msgTypeId,
+            DirectBuffer buffer,
+            int index,
+            int length)
+        {
+            final FrameFW frame = frameRO.wrap(buffer, index, index + length);
+            final long streamId = frame.streamId();
+            final long commonId = commonId(streamId);
```

Comment:
Let's rename this variable to `exchangeId`, as an exchange represents both directions.

Suggest renaming the static method to `initialId` as we are computing the initial id from the reply id when necessary.

Reply:
done